src/backend/adapters/base-record.js

```diff
@@ -63,6 +63,7 @@ class BaseRecord {
     } catch (e) {
       if (e instanceof ValidationError) {
         this.errors = e.errors
+        this.storePayloadData(params)
```

Comment:
I would:
1. rename the method to something like: `storeParams` - describe what the method does from object perspective - only controller knows that this is from payload.
2. put this either in line 62 (before calling to update) or in finally statement. Updating the params is not a special case which happens only in case of error - it should occur always.